ORStools/proc/base_processing_algorithm.py

```diff
@@ -184,9 +187,14 @@ def _get_ors_client_from_provider(
         """
         Connects client to provider and returns a client instance for requests to the ors API
         """
+        current_frame = inspect.currentframe()
+        calling_frame = inspect.getouterframes(current_frame, 2)[1]
+        filename = os.path.basename(calling_frame.filename).split(".")[0]
+        agent = f"QGis_{filename}"
+
```

Comment:
Same as above - it should suffice to use the class name, if we need more info?

Reply:
This way we get a notification for the specific processing algorithm used. Do we just want to differenciate between processing algorithm and GUI?

Reply:
I'm unsure - but either way, we could use the class name, and not add the `os` dependency?

Reply:
Is the os dependency really that big of a deal? I feel like we could use the extra information.

Reply:
I think whenever we can avoid a dependency, we should do so.
Is there any information in the file name that we can't get from the class name?

Reply:
The particular type of processing algorithm used wouldn't be shown, since this is the base processing alg.  Should we pass the filename as a hardcoded variable to the base alg, when using a specific one?

Reply:
Ah, this is a static method, I didn't realize that. We could make this a class method instead, and get the info from the class.